From 258d77921d91b6786706d7ec326c6d9146b4d902 Mon Sep 17 00:00:00 2001
From: Meticulus <theonejohnnyd@gmail.com>
Date: Tue, 14 Nov 2017 04:59:06 -0600
Subject: [PATCH] libhwui: fixes and symbols for libhwuibp

-define out unused code and add missing symbol referenced
by libhwuibp

Change-Id: I5237e41f7155349967d7fc22f769dbce2d9b150e
---
 libs/hwui/ProgramCache.cpp | 4 ++++
 libs/hwui/SpotShadow.cpp   | 2 ++
 2 files changed, 6 insertions(+)

diff --git a/libs/hwui/ProgramCache.cpp b/libs/hwui/ProgramCache.cpp
index f70723c..0bf64b8 100644
--- a/libs/hwui/ProgramCache.cpp
+++ b/libs/hwui/ProgramCache.cpp
@@ -811,3 +811,7 @@ void ProgramCache::printLongString(const String8& shader) const {
 
 }; // namespace uirenderer
 }; // namespace android
+
+extern "C" void _ZN7android10uirenderer12ProgramCache14generateShaderERKNS0_18ProgramDescriptionERNS_7String8ES6_(const android::uirenderer::ProgramDescription& description, android::String8& shader) {
+    PROGRAM_LOGD("Meticulus: '%s'\n", shader.string());
+}
diff --git a/libs/hwui/SpotShadow.cpp b/libs/hwui/SpotShadow.cpp
index cc96a13..d997bc3 100644
--- a/libs/hwui/SpotShadow.cpp
+++ b/libs/hwui/SpotShadow.cpp
@@ -311,9 +311,11 @@ void SpotShadow::makeClockwise(Vector2* polygon, int len) {
     if (polygon == nullptr  || len == 0) {
         return;
     }
+#if 0
     if (!ShadowTessellator::isClockwise(polygon, len)) {
         reverse(polygon, len);
     }
+#endif
 }
 
 /**
-- 
2.7.4

